aero.minova.core.application.system.service/src/test/java/aero/minova/core/application/system/controller/XSqlProcedureControllerTest.java

```diff
@@ -0,0 +1,169 @@
+package aero.minova.core.application.system.controller;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Rule;
+import org.junit.jupiter.api.Test;
+import org.junit.rules.ExpectedException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import aero.minova.core.application.system.domain.Column;
+import aero.minova.core.application.system.domain.DataType;
+import aero.minova.core.application.system.domain.Row;
+import aero.minova.core.application.system.domain.SqlProcedureResult;
+import aero.minova.core.application.system.domain.Table;
+import aero.minova.core.application.system.domain.Value;
+import aero.minova.core.application.system.domain.XSqlProcedureResult;
+import aero.minova.core.application.system.domain.XTable;
+
+//benötigt, damit JUnit-Tests nicht abbrechen
+@SpringBootTest(properties = { "application.runner.enabled=false" })
+public class XSqlProcedureControllerTest extends BaseTest {
+	@Autowired
+	XSqlProcedureController testSubject;
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	@Test
+	public void testFillDependency() {
+		List<XSqlProcedureResult> results = new ArrayList<>();
+
+		Table inputTable = new Table();
+		inputTable.setName("spTest");
+		inputTable.addColumn(new Column("TestText", DataType.STRING));
+		{
+			Row inputRow = new Row();
+			inputRow.addValue(null);
+			inputTable.addRow(inputRow);
+		}
+		SqlProcedureResult sqlres = new SqlProcedureResult();
+		sqlres.setOutputParameters(inputTable);
+		XSqlProcedureResult xRes = new XSqlProcedureResult("test", sqlres);
+		results.add(xRes);
+
+		sqlres.setOutputParameters(inputTable);
+		xRes = new XSqlProcedureResult("test3", sqlres);
+		results.add(xRes);
+
+		// In test2 steht der Value, den wir dann haben wollen.
+		Table testTable = new Table();
+		testTable.setName("xpTest");
+		testTable.addColumn(new Column("TestText", DataType.STRING));
+		{
+			Row inputRow = new Row();
+			inputRow.addValue(new Value("TestString", null));
+			testTable.addRow(inputRow);
+		}
+		sqlres.setOutputParameters(testTable);
+		xRes = new XSqlProcedureResult("test2", sqlres);
+		results.add(xRes);
+
+		// XTable zum Aufrufen der Methode
+		Row inputRow = new Row();
+		inputRow.addValue(new Value("TestText", "test2"));
+		List<Row> rows = new ArrayList<>();
+		rows.add(inputRow);
+		inputTable.setRows(rows);
+
+		XTable xtable = new XTable();
+		xtable.setId("TestTable");
+		xtable.setTable(inputTable);
+
+		Table resTable = testSubject.fillInDependencies(xtable, results);
+
+		assertThat(resTable.getRows().get(0).getValues().get(0).getStringValue()).isEqualTo("TestString");
```

Comment:
Es wurde doch kein Referenz-Wert gesetzt?

Reply:
Doch, es wurde nur keine Row angegeben, in welcher der Wert sein soll.
In dem Fall wird immer von der ersten Row ausgegangen.

Reply:
In Zukunft, sollte die inputTabelle auch von einem JSON-Dokument gelesen werden,
aber es passt erstmals.